tools/src/bin/output-file.rs

```diff
@@ -335,22 +335,58 @@ fn main() {
                     accel_key: None,
                     copyable: false,
                 });
-                if let Some(ref hg_root) = tree_config.paths.hg_root {
+
+                let gh_log_link = tree_config
+                    .paths
+                    .github_repo
+                    .as_ref()
+                    .map(|gh_root| format!("{}/commits/{}/{}", gh_root, oid, encoded_path));
+                let has_gh_log_link = gh_log_link.is_some();
+                if let Some(link) = gh_log_link {
                     vcs_panel_items.push(PanelItem {
-                        title: "Log".to_owned(),
-                        link: format!("{}/log/tip/{}", hg_root, encoded_path),
+                        title: "Git log".to_owned(),
+                        link,
                         update_link_lineno: "",
                         accel_key: Some('L'),
                         copyable: true,
                     });
+                }
+
+                let hg_log_link = tree_config
+                    .paths
+                    .hg_root
+                    .as_ref()
+                    .map(|hg_root| format!("{}/log/tip/{}", hg_root, encoded_path));
+                if let Some(link) = hg_log_link {
+                    vcs_panel_items.push(PanelItem {
+                        title: "Mercurial log".to_owned(),
+                        link,
+                        update_link_lineno: "",
+                        accel_key: (!has_gh_log_link).then_some('L'),
+                        copyable: true,
+                    });
+                }
+
+                let gh_raw_link = tree_config
+                    .paths
+                    .github_repo
+                    .as_ref()
+                    .map(|gh_root| format!("{}/raw/{}/{}", gh_root, oid, encoded_path));
```

Comment:
Same here: should this be `HEAD` instead of  the displayed commit `oid`?

Reply:
Yeah, I think so.  And it's probably more polite load-wise.